### PR TITLE
 Auth - migrate auth.en-marche.fr to en-marche.fr 

### PR DESCRIPTION
**Pull request by B-Galati**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -66,15 +66,25 @@
         "cocur/slugify": "^2.5",
         "symfony/thanks": "^1.0",
         "a2lix/translation-form-bundle": "^2.1",
-        "a2lix/i18n-doctrine-bundle": "^0.1.0"
+        "a2lix/i18n-doctrine-bundle": "^0.1.0",
+        "symfony/thanks": "^1.0",
```
Comment (Remg) [inline composer.json]: duplicated line

File: features/bootstrap/EmailContext.php [hunk 2]
```diff
@@ -0,0 +1,85 @@
+<?php
+
+use AppBundle\Repository\EmailRepository;
+use Behat\Behat\Hook\Scope\BeforeScenarioScope;
+use Behat\Gherkin\Node\PyStringNode;
+use Behat\MinkExtension\Context\RawMinkContext;
+use Behat\Symfony2Extension\Context\KernelDictionary;
+
+class EmailContext extends RawMinkContext
```
Comment (Remg) [inline features/bootstrap/EmailContext.php]: :heart_eyes:

File: features/bootstrap/RabbitMqContext.php [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+
+use Behat\Behat\Context\Context;
+use Behat\Gherkin\Node\TableNode;
+use Coduo\PHPMatcher\PHPMatcher;
+use OldSound\RabbitMqBundle\RabbitMq\Consumer;
+use OldSound\RabbitMqBundle\RabbitMq\Producer;
+use PhpAmqpLib\Message\AMQPMessage;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+class RabbitMqContext implements Context
```
Comment (Remg) [inline features/bootstrap/RabbitMqContext.php]: :heart_eyes:

File: src/Mailer/Message/AdherentResetPasswordConfirmationMessage.php [hunk 4]
```diff
@@ -11,11 +11,11 @@ public static function createFromAdherent(Adherent $adherent): self
     {
         return new self(
             Uuid::uuid4(),
-            '130495',
+            '292297',
             $adherent->getEmailAddress(),
             $adherent->getFullName(),
-            'Confirmation de modification de votre mot de passe',
-            ['target_firstname' => self::escape($adherent->getFirstName())]
+            'Confirmation réinitialisation du mot de passe',
```
Comment (Remg) [inline src/Mailer/Message/AdherentResetPasswordConfirmationMessage.php]: - Confirmation

File: src/Controller/OAuth/OAuthServerController.php [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+            $authRequest->setUser($user->getOAuthUser());
+
+            $client = $repository->findClientByUuid(Uuid::fromString($authRequest->getClient()->getIdentifier()));
+            $form = $this
+                ->createForm(ConfirmActionType::class)
+                ->handleRequest($this->httpFoundationFactory->createRequest($request));
+
+            $authorizationApproved = $manager->isAuthorized($user, $client, $authRequest->getScopes());
+
+            if (
```
[the remark this answers is not in the document]
Comment (B-Galati) [reply src/Controller/OAuth/OAuthServerController.php]: yeap yeap

File: src/Form/UserType.php [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+            ->add('first_name', TextType::class)
+            ->add('last_name', TextType::class)
+            ->add('zip_code', TextType::class)
+            ->add('country', CountryType::class, $this->getCountryOptions($options))
+            ->add('accept_email_notifications', CheckboxType::class, [
+                'required' => false,
+            ])
+        ;
+
+        if (in_array('registration', $options['validation_groups'], true)) {
```
[the remark this answers is not in the document]
Comment (B-Galati) [reply src/Form/UserType.php]: Good catch thanks

[the remark this answers is not in the document]
Comment (B-Galati) [reply src/Form/UserType.php]: In fact this form is not used so I deleted it

File: features/bootstrap/JsonContext.php [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+use Behat\Gherkin\Node\PyStringNode;
+use Behat\Mink\Exception\ExpectationException;
+use Behatch\Context\JsonContext as BehatchJsonContext;
+use Coduo\PHPMatcher\PHPMatcher;
+
+class JsonContext extends BehatchJsonContext
+{
+    public function theJsonShouldBeEqualTo(PyStringNode $content)
+    {
+        $actual = $this->getJson();
```
Comment (thomasbisignani) [inline features/bootstrap/JsonContext.php]: This declaration variable is not really needed

File: src/Entity/OAuth/AbstractGrantToken.php [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+        \DateTime $expiryDateTime,
+        Client $client = null
+    ) {
+        parent::__construct($uuid, $identifier, $expiryDateTime);
+
+        $this->user = $user;
+        $this->client = $client;
+    }
+
+    public function getUser(): Adherent
```
[the remark this answers is not in the document]
Comment (B-Galati) [reply src/Entity/OAuth/AbstractGrantToken.php]: I'll fix that in the webhook Pr ;-)

File: features/bootstrap/JsonContext.php (hunk 7 above)
Comment (B-Galati) [reply features/bootstrap/JsonContext.php]: agree thank you. I'll fix it in WebHook Pr.

**Pull request merged**